kani-driver/src/cbmc_output_parser.rs

```diff
@@ -420,6 +433,24 @@ impl<'a, 'b> Parser<'a, 'b> {
         if let Ok(item) = result_item {
             return item;
         }
+        // If we failed to parse a `ParserItem::Result` earlier, we will get
+        // this error message when we attempt to parse it using the complete
+        // string:
+        // ```
+        // thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value:
+        // Error("data did not match any variant of untagged enum ParserItem", line: 0, column: 0)'
+        // ```
+        // This error message doesn't provide information about what went wrong
+        // while parsing due to `ParserItem` being an untagged enum. A more
+        // informative error message will be produced if we attempt to
+        // deserialize it into a struct. The attempt will still fail, but it
+        // shouldn't be hard to debug with that information. The same strategy
+        // can be used for other `ParserItem` variants, but they're normally
+        // easier to debug.
+        if string_without_delimiter.starts_with(RESULT_ITEM_PREFIX) {
```

Comment:
Have you considered changing the signature of this function to Result<ParserItem> and in the case of failure you can add the context that we failed on (i.e., what was the item that we were trying to parse).

Reply:
We don't know which item we're trying to parse, that's why we use the `ParserItem` enum. I can change the signature to use `Result<ParserItem>` but I don't know what to with the result other than `unwrap()`.

Reply:
I made the change but then dropped it because the new case would return a `ResultStruct`. I'm open to suggestions here.

Reply:
Could you use `expect()` and add to the error message the string that we failed to parse.

Reply:
Thanks for explaining the issue offline. I'll resolve this comment.

Reply:
I cannot get rid of `#[allow(unused)]` even if use `ResultStruct.result` in the test.